### PR TITLE
Fix some UI issues in style manager

### PR DESCRIPTION
**Pull request by SebDieBln**
While fixing some bugs in style manager I stumbled upon a few UI issues:
- minor **memory leaks** caused by orphaned `QMenu`s and `QAction`s
- **labels**
  - not following the guidelines regarding capitalization or usage of '...'
  - being unclear
  - denominating the wrong entity type (_style_ instead of _symbol_)
  - being different for identical action
- **actions**
  - still enabled when not available
  - missing in context menu

### CODE REVIEW

File: python/core/symbology-ng/qgsstylev2.sip [hunk 1]
```diff
@@ -270,6 +275,9 @@ class QgsStyleV2 : QObject
     //! gets the id from the table for the given name from the database, 0 if not found
     int getId( const QString& table, const QString& name );
 
+    //! gets the name from the table for the given id from the database, QString::null if not found
```
Comment (nyalldawson) [inline python/core/symbology-ng/qgsstylev2.sip]: @SebDieBln note that Qt discourage use of null strings - see http://doc.qt.io/qt-4.8/qstring.html#distinction-between-null-and-empty-strings . I suspect based on this that in some future Qt version null strings will be removed. It's safer to use an empty string instead.

**Comment by nyalldawson** [comment]
@SebDieBln seriously great stuff - very nice! 

One last possible tweak would be renaming "Un-group" to "Ungroup". That would keep consistency with composer and the "Ungrouped" style group.

**Comment by NathanW2** [comment]
Nice clean up work.

File: python/core/symbology-ng/qgsstylev2.sip (hunk 1 above)
Comment (SebDieBln) [inline python/core/symbology-ng/qgsstylev2.sip]: Thanks for the tip!
I remember reading these pages a while ago, noting the subtle difference between `QString::isNull()` and `QString::isEmpty()`. But the discouragement of `QString::null` somehow did not catch my attention.

**Comment by SebDieBln** [comment]
@nyalldawson 
> One last possible tweak would be renaming "Un-group" to "Ungroup".

I Changed it. Consistency rocks!

**Comment by nirvn** [comment]
Nice work.

You know what would be nice of the style manager? For it to be a floating dockable panel, allowing for drag n dropping of styles from the manager onto a layer to style the latter. :)

File: src/core/symbology-ng/qgsstylev2.cpp [hunk 2]
```diff
@@ -947,6 +961,24 @@ int QgsStyleV2::getId( const QString& table, const QString& name )
...
 }
 
+QString QgsStyleV2::getName( const QString& table, int id ) const
+{
+  char *query = sqlite3_mprintf( "SELECT name FROM %q WHERE id='%q'", table.toUtf8().constData(), QString::number( id ).toUtf8().constData() );
+
+  sqlite3_stmt *ppStmt;
+  int nErr = sqlite3_prepare_v2( mCurrentDB, query, -1, &ppStmt, nullptr );
+
+  QString name = "";
```
Comment (nyalldawson) [inline src/core/symbology-ng/qgsstylev2.cpp]: Don't do this either ;) it means the literal has to be parsed into the string. Just using the default constructor has the same effect but is more efficient.

**Comment by SebDieBln** [comment]
> allowing for drag n dropping of styles

One thing I did not fix is the somewhat misleading name __Style__ manager. I think it should be __Symbol__ manager, because it manages symbols, not styles.

What should happen, when the user drops a single symbol onto a layer styled with a categorized or rule based style?

Drag and drop would be nice whereever a symbol is needed. But that is a different PR.

**Comment by nirvn** [comment]
@SebDieBin, yeah, I wasn't mentioning this in relation to this PR, simply planting the seed so it becomes reality one day :)

For rule-based / categorized / graduated symbology, the saved symbol would have to be dropped onto a child node of the layer.

**Comment by SebDieBln** [comment]
> For rule-based / categorized / graduated symbology, the saved symbol would have to be dropped onto a child node of the layer.

Definitly :+1: for that!

**Comment by nyalldawson** [comment]
> One thing I did not fix is the somewhat misleading name Style manager. I think it should be Symbol manager, because it manages symbols, not styles.

I think "Style" is correct here. It also manages ramps, which aren't symbols. And it also covers us if in future labels styles or something similar is added.

File: src/core/symbology-ng/qgsstylev2.cpp (hunk 2 above)
Comment (SebDieBln) [inline src/core/symbology-ng/qgsstylev2.cpp]: Efficiency rocks, too!

**Comment by nyalldawson** [comment]
Perfect! I'll merge as soon as it passes the Travis linux build (seems to be back up). OSX build has unrelated issues.

Here's some more UI issues in this dialog which might be nice to address:
- add separators to the context menu (eg group, ungroup, separator, edit/remove, seperator, export options)
- add a confirm messagebox before removing symbols. I just accidentally deleted a bunch when testing this ;)

**Comment by SebDieBln** [comment]
What actually bothers me is the ambiguity of the term _style_.
* Looking at the source, the style manager is initialized with a `QgsStyleV2* style`, meaning a set of symbols and ramps.
* A _style_ is also something that defines, how a layer is rendered, including labels, scale based visibility and other settings. This can be saved to a QML file.
* And _style_ is also used as the name of a tab within the layer properties.

But I expected a discussion about this term and therefore left it out of this PR on purpose :smiley:

**Comment by nyalldawson** [comment]
Also I think the "Edit symbol" action should be "Edit symbol...", and also the Export as PNG/SVG actions should have "..."

**Comment by SebDieBln** [comment]
@nyalldawson Good points. I will add them tomorrow. So please do not merge yet.

**Comment by SebDieBln** [comment]
I added
* separators in the context menu
* '...' for _edit item_ and _export as PNG/SVG_
* confirmation dialogs before removing items

Note that _master_ is still broken at the moment (bd9d41c6eefeaeeda337f70bd6ff2fc55c85b3d5), so could someone please re-trigger the build in Travis when master is fixed? In the meantime please see the [build of the branch in my fork](https://travis-ci.org/SebDieBln/QGIS/builds/100711621).

**Comment by nyalldawson** [comment]
Perfect - fantastic work! Love it.

**Pull request merged**